Diggy/Feature/DiggyTilePressure.lua

```diff
@@ -0,0 +1,52 @@
+--[[-- info
+    Provides the ability to collapse caves when digging.
+]]
+
+-- dependencies
+
+local Event = require 'utils.event'
+local PressureMap = require 'Diggy.PressureMap'
+
+-- this
+local DiggyTilePressure = {}
+
+
+
+--[[--
+    Registers all event handlers.]
+
+    @param config Table {@see Diggy.Config}.
+]]
+function DiggyTilePressure.register(config)
+    Event.add(PressureMap.events.on_pressure_changed, function(event)
+      
+        local e = game.surfaces['nauvis'].find_entity("flying-text", event.position)
+        if e then   e.destroy() end
+        
+        local r = event.value
+        local g = 1 - event.value
+        if r < 0 then r = 0 end
+        if r > 1 then r = 1 end
+        if g < 0 then g = 0 end
+        if g > 1 then g = 1 end
+        
+        local e = game.surfaces['nauvis'].create_entity{

```

Comment:
This will cause issues on a different surface. `on_pressure_changed` should probably receive the surface. The pressure map is indexed by the surface, so the easiest fix would be to add it as property to this map

Reply:
Yes i realized that. however since this was mainly used for debugging now i did it this way, since as you pointed out there was no way back from map ==> surface.